src/main/java/plugily/projects/villagedefense/events/Events.java

```diff
@@ -433,39 +435,48 @@ public void onCraft(CBPlayerInteractEvent event) {
   }
 
   @EventHandler
-  public void onRottenFleshDrop(InventoryPickupItemEvent e) {
+  public void onSecretWellDrop(InventoryPickupItemEvent e) {
     if(e.getInventory().getType() != InventoryType.HOPPER) {
       return;
     }
-    if(e.getItem().getItemStack().getType() != Material.ROTTEN_FLESH) {
-      for(Arena arena : ArenaRegistry.getArenas()) {
-        if(e.getItem().getWorld().equals(arena.getStartLocation().getWorld())) {
-          e.getItem().remove();
-          e.getInventory().clear();
-          return;
-        }
+    Arena currentArena = null;
+    for(Arena arena : ArenaRegistry.getArenas()) {
+      if(e.getItem().getWorld().equals(arena.getStartLocation().getWorld())) {
+        currentArena = arena;
+        e.getItem().remove();
```

Comment:
The item is removed here, so in the `getNearbyEntities` loop the location of this item will be null or unloaded.